cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java

```diff
@@ -272,45 +308,53 @@ private Workflow compileWorkFlow(List<Map<String, Map<String, Object>>> workFlow
             } else {
                 defaultSuccess = onFailureSection ? ScoreLangConstants.FAILURE_RESULT : ScoreLangConstants.SUCCESS_RESULT;
             }
-            Task task = compileTask(taskName, taskRawDataValue, defaultSuccess, imports, defaultFailure, namespace);
-            tasks.add(task);
+            TaskModellingResult taskModellingResult = compileTask(taskName, taskRawDataValue, defaultSuccess, imports, defaultFailure, namespace);
+            errors.addAll(taskModellingResult.getErrors());
+            tasks.add(taskModellingResult.getTask());
         }
 
         if (isOnFailureDefined) {
             tasks.addAll(onFailureWorkFlow.getTasks());
         }
 
-        return new Workflow(tasks);
+        return new WorkflowModellingResult(new Workflow(tasks), errors);
     }
 
-    private Task compileTask(String taskName, Map<String, Object> taskRawData, String defaultSuccess,
-                                     Map<String, String> imports, String defaultFailure, String namespace) {
+    private TaskModellingResult compileTask(String taskName, Map<String, Object> taskRawData, String defaultSuccess,
+                                            Map<String, String> imports, String defaultFailure, String namespace) {
 
+        List<RuntimeException> errors = new ArrayList<>();
         if (MapUtils.isEmpty(taskRawData)) {
-            throw new RuntimeException("Task: " + taskName + " has no data");
+            taskRawData = new HashMap<>();
+            errors.add(new RuntimeException("Task: " + taskName + " has no data"));
         }
 
         Map<String, Serializable> preTaskData = new HashMap<>();
         Map<String, Serializable> postTaskData = new HashMap<>();
 
-        transformersHandler.validateKeyWords(taskName, taskRawData, ListUtils.union(preTaskTransformers, postTaskTransformers), TaskAdditionalKeyWords, null);
+        errors.addAll(transformersHandler.checkKeyWords(taskName, taskRawData, ListUtils.union(preTaskTransformers, postTaskTransformers), TaskAdditionalKeyWords, null));
 
-        try {
-            preTaskData.putAll(transformersHandler.runTransformers(taskRawData, preTaskTransformers));
-            postTaskData.putAll(transformersHandler.runTransformers(taskRawData, postTaskTransformers));
-        } catch (Exception ex){
-            throw new RuntimeException("For task: " + taskName + " syntax is illegal.\n" + ex.getMessage(), ex);
-        }
+        String errorMessagePrefix = "For task: " + taskName + " syntax is illegal.\n";
```

Comment:
I see that you specify a prefix for tasks but you don't do that for executable and action - Y?